cmd/atlas/atlas.go

```diff
@@ -32,7 +32,11 @@ import (
 func Execute() {
 	ctx := telemetry.NewContext()
 	rootCmd := atlas.Builder()
-	if err := rootCmd.ExecuteContext(ctx); err != nil {
+	if cmd, err := rootCmd.ExecuteContextC(ctx); err != nil {
+		if !telemetry.StartedTrackingCommand() {
+			telemetry.StartTrackingCommand(cmd, os.Args[1:])
+		}
```

Comment:
should this be moved outside of the if statement? start track -> execute -> finish track

Reply:
I don't follow

Reply:
`telemetry.StartTrackingCommand` is called in `PersistentPreRunE` from `rootCmd`, but when errors like `clister ls command not found` happen `PersistentPreRunE` never fires.

But on the other hand, some errors like `authentication expired` the prerun would have been triggered since the command itself ran and failed in its execution.

Reply:
tracking is started here after the command is executed, not before, is this intentional or should it be like this:
```
    if !telemetry.StartedTrackingCommand() {
        telemetry.StartTrackingCommand(cmd, os.Args[1:])
    }
    if cmd, err := rootCmd.ExecuteContextC(ctx); err != nil {
        telemetry.FinishTrackingCommand(telemetry.TrackOptions{
            Err: err,
        })
    }
```